frontend/src/components/HomePage/index.jsx

```diff
@@ -1,74 +1,97 @@
+  ],
+  "Government": [
+    "legislature", "executive", "judiciary", "administration", "policy",
+    "governance", "legislation", "regulation", "budget", "oversight",
+    "authority", "accountability", "federal", "state", "local", "municipal",
+    "council", "mayor", "governor", "president", "congress", "senate", "house",
+    "law", "regulation", "committee", "commission", "agency", "department",
+    "bureau", "office", "public", "official", "civil", "servant", "leadership",
+    "rule", "decision", "policy", "legislation", "governmental", "civic",
+    "constitution", "statute", "ordinance", "charter", "administration",
+    "judiciary"
+  ],
+  "Healthcare": [
+    "healthcare", "health", "medical", "hospital", "doctor", "patient",
+    "nurse", "clinic", "medicine", "insurance", "provider", "treatment",
+    "wellness", "pharmacy", "coverage", "vaccine", "disease", "emergency",
+    "public health", "health system", "pandemic", "care", "mental health"
+  ]
+};
+
+// Function to get up to 3 categories for a bill based on keyword count
+const getCategories = (summary) => {
+  const matchingCategories = Object.entries(categories)
+    .filter(([category, keywords]) => keywords.some(keyword => summary.toLowerCase().includes(keyword)))
+    .map(([category]) => category);
+  return matchingCategories.slice(0, 3);
+};
+
+
 const HomePage = () => {
   const [bills, setBills] = useState([]);
-  const [categoryMappings, setCategoryMappings] = useState({});
-  const [categories, setCategories] = useState({});
 
   useEffect(() => {
-
-    // Correctly handle double-mount in dev/StrictMode
-    // https://stackoverflow.com/a/72238236
     const abortController = new AbortController();
 
-    // TODO: does not fetch all bills, only first page
-    const paginateBills = async () => {
-      let offset = 1;
-      let done = false;
-      while (!done) {
-        try {
-          const res = await fetch(`/api/v1/bills/2023/search?offset=${offset}`, {
-            signal: abortController.signal,
-          });
-          const out = await res.json();
-          await setBills((prevBills) => [...prevBills].concat(out.result.items.map(item => item.result)));
-          offset = out.offsetEnd;
-          // if (out.offsetEnd >= out.total) done = true;
-          done = true;
-        } catch (error) {
-          console.log('Home bills request aborted');
-          done = true;
-        }
-      }
-    };
-
-    const billCategoryMappings = async () => {
+    const fetchData = async () => {
       try {
-        const res = await fetch(`/api/v1/bills/category-mappings`, {
-          signal: abortController.signal,
-        });
-        setCategoryMappings(await res.json());
+        const res = await fetch(`/api/v1/bills/2023/search`, { signal: abortController.signal });
+        const data = await res.json();
+        // Assign categories to each bill
+        const billsWithCategories = data.result.items.map(item => ({
```

Comment:
I'm not a fan of merging our data with existing data.
- makes the object less predictable & less documented
- makes this initial fetch take more time

I'd suggest dynamically determining this in the homepage `Card`.

Reply:
totally fair, and good points. I did do that initially but the css wasn't cooperating. I'll focus on this tonight as well